receiver/zipkinreceiver/config.go

```diff
@@ -4,22 +4,71 @@
 package zipkinreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/zipkinreceiver"
 
 import (
+	"fmt"
+
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/featuregate"
+)
+
+var disallowHTTPDefaultProtocol = featuregate.GlobalRegistry().MustRegister(
+	"zipkinreceiver.httpDefaultProtocol.disallow",
+	featuregate.StageAlpha,
+	featuregate.WithRegisterDescription("When enabled, usage of the default http configuration is disallowed"),
+	featuregate.WithRegisterFromVersion("v0.114.0"),
 )
 
+const deprecationConfigMsg = "the inline setting of http server parameters has been deprecated, please use .protocols.http parameter instead."
+
 // Config defines configuration for Zipkin receiver.
 type Config struct {
 	// Configures the receiver server protocol.
+	//
+	// Deprecated: Parameter exists for historical compatibility
+	// and should not be used. To set the server configurations,
+	// use the Protocols parameter instead.
 	confighttp.ServerConfig `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
 	// If enabled the zipkin receiver will attempt to parse string tags/binary annotations into int/bool/float.
 	// Disabled by default
 	ParseStringTags bool `mapstructure:"parse_string_tags"`
+	// Protocols define the supported protocols for the receiver
+	Protocols ProtocolTypes `mapstructure:"protocols"`
+}
+
+type ProtocolTypes struct {
+	HTTP confighttp.ServerConfig `mapstructure:"http"`
 }
 
 var _ component.Config = (*Config)(nil)
 
 // Validate checks the receiver configuration is valid
 func (cfg *Config) Validate() error {
+	if isServerConfigDefined(cfg.ServerConfig) {
+		if disallowHTTPDefaultProtocol.IsEnabled() {
+			return fmt.Errorf(deprecationConfigMsg)
```

Comment:
I'm not sure if re-using the deprecation warning message as the error message when the feature gate is enabled will be clear enough for the users when they see it.